EIPS/eip-7904.md

```diff
@@ -0,0 +1,406 @@
+
+Projects like Nethermind's Gas Benchmarks highlight a security concern: lowering the gas cost for ECRECOVER could lead to the worst-case scenario where the block computation time exceeds safety threshold. A similar issue applies to MODEXP. As the result, the `rescale factor` cannot be lower that the proposed `0.217391304` even though there is a room for further reduction. Note that the maximum number of operations in a block depends on the gas cost schedule and the block gas limit.
+
+The Gas Cost Estimator project suggests a slight increase in ECRECOVER’s gas cost, but this proposal avoids that change to maintain backward compatibility and because the impact would be minor.
+
+For hashing operations, this proposal covers the precompiles SHA2-256, RIPEMD-160, BLAKE2F, and the KECCAK256 opcode. Key updates include:
+
+- The gas cost for BLAKE2F remains unchanged.
+- The per-word gas cost for KECCAK256 stays the same.
+- For SHA2-256 and RIPEMD-160, the per-word gas cost is reduced by a factor of 3.
+
+[EIP-7667](./eip-7667.md) examines worst-case scenarios for generating ZK proofs in blocks filled entirely with hashing operations. Although this reduction for SHA2-256 and RIPEMD-160 might seem substantial, EIP-7667 takes a more conservative approach. Therefore, the changes proposed here do not worsen the edge cases outlined in that EIP.
+
+## Backwards Compatibility
+
+The proposed changes to the gas cost schedule will require a hardfork due to their significant impact on the Ethereum network's operational mechanics. Below, we outline the key consequences of these changes and highlight areas where further research is needed to ensure a smooth transition.
+
+The changes have the following consequences:
+
+- Gas Cost Adjustments:
+  - The gas costs for a wide range of opcodes, precompiles, and other operations (such as memory expansion and access costs) will be modified.
+  - These adjustments aim to better align gas pricing with the computational complexity of operations, but they will directly impact the cost of executing transactions and smart contracts.
+- Transaction Gas Cost Changes:
+  - It is highly likely that the gas cost of any transaction that invokes a contract will change.
+  - This is due to the widespread use of affected opcodes and operations within smart contract code.
+  - Developers and users should prepare for potential variations in transaction fees after the hardfork, which may require updates to gas limit estimations and fee budgeting.
+- Impact on Contracts with Hardcoded Gas Limits:
+  - Contracts that specify hardcoded gas limits for subcalls (e.g., using call(gas, ...)) may face issues if the new gas costs exceed these limits.
+  - Such contracts could fail to execute as intended, potentially resulting in transaction failures or unexpected behavior.
+
+Further research might be required to ensure that contracts that use hard coded limits are not broken.
+
+## Test Cases
+
+### Test 1
+
+Code:
```

Comment:
are testcases cover all proposed changes? if not add <-- TODO --> here or add links to the test suites which are accepted by the EIP repo

Reply:
lemme know your response over this

Reply:
Hey, I've added the following comment to the EIP:
```
TODO: This EIP is a relatively simple change, but requires updates in many tests (spec, execution API, etc.). Due to the effort needed to update the tests, this will be provided only after the EIP is accepted.
```

Reply:
thats alright, but could you add the html TODO marker which the bot will auto flag if you try to move it past review without addressing the comment

Reply:
Done